daffodil-lib/src/main/scala/org/apache/daffodil/util/RangeBound.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.util
+
+import java.lang.{ Long => JLong }
+import org.apache.daffodil.exceptions.Assert
+import org.apache.daffodil.cookers.Converter
+import org.apache.daffodil.exceptions.ThrowsSDE
+import org.apache.daffodil.util.RangeBound.Range
+
+object RangeBound {
```

Comment:
This object is the only real change in this file. Otherwise, it is just a direct move from the typeCalculator file.